lib/core/viewmodels/not_found_viewmodel.dart

```diff
@@ -0,0 +1,61 @@
+// FLUTTER / DART / THIRD-PARTIES
+import 'package:flutter/material.dart';
+import 'package:rive/rive.dart';
+import 'package:stacked/stacked.dart';
+
+// SERVICE
+import 'package:notredame/core/services/analytics_service.dart';
+import 'package:notredame/core/services/navigation_service.dart';
+import 'package:notredame/core/services/rive_animation_service.dart';
+
+// OTHER
+import 'package:notredame/locator.dart';
+import 'package:notredame/core/constants/router_paths.dart';
+
+class NotFoundViewModel extends BaseViewModel {
+  static const String tag = "NotFoundViewModel";
+
+  /// Used to redirect on the dashboard.
+  final NavigationService _navigationService = locator<NavigationService>();
+  //Used to log the event that pushedit from
+  final AnalyticsService _analyticsService = locator<AnalyticsService>();
+  //Used to log the event that pushedit from
+  final RiveAnimationService _riveAnimationService =
+      locator<RiveAnimationService>();
+
+  final String _riveAnimationFileName = 'dot_jumping';
+
+  final String _pageName;
+  String get notFoundPageName => _pageName;
```

Comment:
Because the variable is final, there is no need to create a getter function, if you need to access it from outside the view-model just make the variable public 😄